From 0cb1520f87ef4db206d0bc3c523a61303d5061a3 Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:24 +0800
Subject: [PATCH] sw64: clean up out-of-date selected options
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

These options have been deprecated for a long time.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/Kconfig | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/arch/sw_64/Kconfig b/arch/sw_64/Kconfig
index cf1d4b97835b..4c43462b6584 100644
--- a/arch/sw_64/Kconfig
+++ b/arch/sw_64/Kconfig
@@ -3,24 +3,17 @@ config SW64
 	bool
 	default y
 	select AUDIT_ARCH
-	select VIRT_IO
-	select HAVE_AOUT
 	select HAVE_IDE
 	select HAVE_OPROFILE
-#	select HAVE_SYSCALL_WRAPPERS
-	select HAVE_IRQ_WORK
 	select HAVE_PCSPKR_PLATFORM
 	select HAVE_PERF_EVENTS
-	select HAVE_GENERIC_HARDIRQS
 	select HAVE_GENERIC_GUP
 	select GENERIC_CLOCKEVENTS
 	select GENERIC_IRQ_PROBE
 	select GENERIC_IRQ_LEGACY
-	select GENERIC_IDLE_LOOP
 	select GENERIC_IRQ_SHOW
 	select ARCH_WANT_IPC_PARSE_VERSION
 	select ARCH_HAVE_NMI_SAFE_CMPXCHG
-	select ARCH_SUPPORTS_MSI
 	select ARCH_MIGHT_HAVE_PC_SERIO
 	select ARCH_NO_PREEMPT
 	select ARCH_USE_CMPXCHG_LOCKREF
@@ -28,12 +21,9 @@ config SW64
 	select HAVE_MOD_ARCH_SPECIFIC
 	select MODULES_USE_ELF_RELA
 	select ARCH_SUPPORTS_NUMA_BALANCING
-	select ARCH_WANTS_PROT_NUMA_PROT_NONE
 	select HAVE_ARCH_TRANSPARENT_HUGEPAGE
-	select HAVE_GENERIC_RCU_GUP
 	select HAVE_ARCH_AUDITSYSCALL
 	select HAVE_ARCH_SECCOMP_FILTER
-	select GENERIC_SIGALTSTACK
 	select OLD_SIGACTION
 	select OLD_SIGSUSPEND
 	select GENERIC_STRNCPY_FROM_USER
-- 
2.34.1

